aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/subscription/mqtt/MqttSubscriptionClient.java

```diff
@@ -138,8 +138,19 @@ public void setTransmitting(final boolean isTransmitting) {
     public void close() {
         final String clientRepresentation = mMqttAndroidClient.toString();
         try {
-            mMqttAndroidClient.disconnect();
-            mMqttAndroidClient.close();
+            //Disconnect the connection with quiese timeout set to 0, which means disconnect immediately.
+            //Issue the close connection on the callback - this ensures that the connection is properly closed out before resources are freed/reclaimed by the close method.
+            mMqttAndroidClient.disconnect(0,null, new IMqttActionListener() {
+                        @Override
+                        public void onSuccess(IMqttToken asyncActionToken) {
+                            mMqttAndroidClient.close();
+                        }
+
+                        @Override
+                        public void onFailure(IMqttToken asyncActionToken, Throwable exception) {
+
```

Comment:
Probably should add a Log warning/ error message here.

Reply:
Good idea. Will do.